common/reactive/src/main/java/io/helidon/common/reactive/Multi.java

```diff
@@ -843,9 +843,13 @@
      *
      * @param consumer consumer to be invoked for each item
      */
-    default void forEach(Consumer<? super T> consumer) {
-        FunctionalSubscriber<T> subscriber = new FunctionalSubscriber<>(consumer, null, null, null);
+    default MultiCompletionStage forEach(Consumer<? super T> consumer) {
```

Comment:
I'd definitely add cancellation support to this. `AutoCloseable` perhaps.

Reply:
That's pretty cool! :+1: